test/e2e/self_hosted_test.go

```diff
@@ -36,9 +36,12 @@ import (
 )
 
 func TestSelfHosted(t *testing.T) {
-	t.Run("self hosted", func(t *testing.T) {
-		t.Run("create self hosted cluster from scratch", testCreateSelfHostedCluster)
-		t.Run("migrate boot member to self hosted cluster", testCreateSelfHostedClusterWithBootMember)
+	t.Run("create self hosted cluster from scratch", testCreateSelfHostedCluster)
+	t.Run("migrate boot member to self hosted cluster", testCreateSelfHostedClusterWithBootMember)
+	t.Run("backup for self hosted cluster", func(t *testing.T) {
```

Comment:
The cleanup function is not simple.
It sets up pods on each node, runs commands, waits them to complete. It's not worth to do it again just for code structure.